Added maze_with_rooms map

Ogg-Technologies opened the pull request:
Added a new map which makes everything void except a random maze
Randomly sized rooms are placed around the maze

Ogg-Technologies committed:
Added maze_with_rooms map

Added a new map which makes everything void except a random maze
Randomly sized rooms are placed around the maze

grilledham commented on map_gen/maps/maze_with_rooms.lua:
```diff
@@ -0,0 +1,354 @@
+-- Goes through the map and finds dead ends (tiles with 3 walls around them) and fills them in
+-- does this recursivly until every tile has at least 2 ground neighbours
+local function remove_dead_ends(map)
+    function get_neighbours_with_ground(x, y)
```
missing local

Ogg-Technologies replied:
Good catch

grilledham commented on map_gen/maps/maze_with_rooms.lua:
```diff
@@ -0,0 +1,354 @@
+        end
+        return neighbours_with_ground
+    end
+    function remove_dead_end(x, y)
```
missing local.
Also maybe this function should have a different name to the containing function, maybe remove_dead_end_inner, naming is hard.

Ogg-Technologies replied:
I don't really like adding inner after it. The outer function (remove_dead_ends) loops through all tiles that could be dead ends and then calls remove_dead_end on them. But I agree that having such similar names is confusing. Maybe the outer could be remove_all_dead_ends and inner could be fill_dead_end

grilledham commented on map_gen/maps/maze_with_rooms.lua:
```diff
@@ -0,0 +1,354 @@
+local Global = require 'utils.global'
+local RS = require 'map_gen.shared.redmew_surface'
+local b = require 'map_gen.shared.builders'
+local Event = require 'utils.event'
```
We normally put all our requires at the top of the file, so we can easily see the dependencies.

Ogg-Technologies replied:
Yeah, the reason I didn't was that when I tested it I wanted to run the file separatly and print out the map. So I put all factorio related stuff at the same place so I could comment it out when testing. But I can move it now since it works.

grilledham reviewed (approved):
This looks good to me.
If you are interested in micro optimisations I can leave a few more suggestions.

Ogg-Technologies committed:
Small refactoring

Added missing local to functions

Moved requires to the top

Removed unused Event require

Renamed remove_dead_ends functions

Ogg-Technologies committed:
Added ores at dead ends

Ogg-Technologies committed:
Fixed reloading

grilledham commented:
What if any are the changes to this map since we last played it on the Redmew server?

Ogg-Technologies commented:
I haven't made any changes since we played it on Redmew. This should be the same code. Although I sent the code to Jayefuu who set up the map and may have tweaked some constants but from what I know this is the same code.

grilledham commented:
Ok, we should merge this the. @Ogg-Technologies is that ok with you?

Ogg-Technologies commented:
Yeah, merge it

grilledham commented:
Cool, thanks for your contribution. 👍
